feat(ocaml): add language

edwintorok opened the pull request:
Uses https://ocaml.org/p/ocaml-lsp-server/ as language server that can
be installed via `opam`.

Can be tested if I add this to the Dockerfile:
```diff
diff --git a/tools/Dockerfile b/tools/Dockerfile
index 5f2d7ce..e0bc98e 100644
--- a/tools/Dockerfile
+++ b/tools/Dockerfile
@@ -36,6 +36,9 @@ RUN npm i -g chokidar-cli
 # For startup time profiling
 RUN pacman -Sy hyperfine --noconfirm
 
+# OCaml
+RUN pacman -Sy opam diffutils patch ocaml --noconfirm
+
 # Create the doom user and group
 RUN groupadd doom
 RUN useradd -m -g doom doom
@@ -46,6 +49,8 @@ RUN chown -R ${UNAME}:${GNAME} /usr/local/lib/node_modules/
 USER doom
 WORKDIR /home/doom
 
+RUN opam init --disable-sandboxing && opam install ocaml-lsp-server -y
+
 COPY _docker_entry.sh /usr/local/bin/
 
 # Doom-nvim-contrib
```

And then run `eval $(opam env); nvim` inside.

I've been using this for quite a while on 3.3.0, but the port to 4.0.1 is still fresh and needs more testing, hence opening as a draft PR.
Please feel free to point out if this is not the way to add a new language.

connorgmeehan commented:
Hey @edwintorok thanks for the PR.  This looks perfect, I'll have to test it after work but thank you for the PR :)

connorgmeehan commented:
Just tested it and everything works well for me.  Could you add 
```diff
diff --git a/tools/Dockerfile b/tools/Dockerfile
index 5f2d7ce..e0bc98e 100644
--- a/tools/Dockerfile
+++ b/tools/Dockerfile
@@ -36,6 +36,9 @@ RUN npm i -g chokidar-cli
 # For startup time profiling
 RUN pacman -Sy hyperfine --noconfirm
 
+# OCaml
+RUN pacman -Sy opam diffutils patch ocaml --noconfirm
+
 # Create the doom user and group
 RUN groupadd doom
 RUN useradd -m -g doom doom
@@ -46,6 +49,8 @@ RUN chown -R ${UNAME}:${GNAME} /usr/local/lib/node_modules/
 USER doom
 WORKDIR /home/doom
 
+RUN opam init --disable-sandboxing && opam install ocaml-lsp-server -y
+
 COPY _docker_entry.sh /usr/local/bin/
 
 # Doom-nvim-contrib
```
to the `Dockerfile` but leave the changes commented out and leave a message `# Required for ocaml language`?  I think it's worth keeping it in but commented out because it's useful knowledge.  Once that's done happy to merge :)

edwintorok commented:
Thanks, updated the PR (also rebased on top of latest main). BTW the docs say I should create the PR targeting the `develop` branch, but that branch seems to be way behind, is there another branch I should target instead for new features?

connorgmeehan commented:
Thanks @edwintorok! I appreciate these up-streams :)
